internal/openstack/client.go

```diff
@@ -154,3 +165,31 @@ func (client *OpenStackClient) NewNetworkV2() (*gophercloud.ServiceClient, error
 		Region: client.region,
 	})
 }
+
+// Use the project name and project domain ID from the auth options to get the project ID.
+func getProjectID(provider *gophercloud.ProviderClient, cfg *AuthOpts) (string, error) {
+	identityClient, err := openstack.NewIdentityV3(provider, gophercloud.EndpointOpts{Region: cfg.Region})
+	if err != nil {
+		return "", err
+	}
+
+	projectPage, err := projects.List(
+		identityClient,
+		projects.ListOpts{DomainID: cfg.ProjectDomainID, Name: cfg.ProjectName},
+	).AllPages()
```

Comment:
```
F1102 13:20:06.868383       1 ch-k8s-lbaas-controller.go:84] Failed to connect to OpenStack: Request forbidden: [GET https://identity-f1a.cloudandheat.com:5000/v3/projects?domain_id=default&name=cah-jonasschaefer-k8s-dev], error message: {"error":{"code":403,"message":"You are not authorized to perform the requested action: identity:list_projects.","title":"Forbidden"}}
```

This seems to require admin privileges. The equivalent of `openstack project show $name` should work though, because our token is scoped to the domain implicitly by being scoped to the project already (hopefully).

Reply:
Turns out, `openstack project show $OS_PROJECT_NAME` actually smarts out and uses information from the token to sidestep the permission issue.

And `openstack project list` uses the user's project listing (separate API endpoint) using the user ID obtained from the token.

Neither options are available from gophercloud trivially.

Reply:
Following the example from https://github.com/gophercloud/gophercloud/blob/master/auth_result.go, I now extract the project ID from gopherclouds stored authentication result.
This should work for all users.

Reply:
Thanks!